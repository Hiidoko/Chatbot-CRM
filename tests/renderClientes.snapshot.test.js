/** @jest-environment jsdom */
const { renderClientes } = require('../public/js/crm/dom.cjs');

function mockCliente(overrides = {}) {
  return {
    id: 1,
    nome: 'Cliente Teste',
    email: 'cliente@teste.com',
    telefone: '(11) 99999-9999',
    cidade: 'São Paulo',
    maquina: 'Máquina A',
    horario: 'Manhã',
    status: 'novo',
    consultor: 'Consultor X',
    dataCadastro: new Date('2024-01-01').toISOString(),
    ...overrides
  };
}

describe('renderClientes (estrutura semântica)', () => {
  test('gera <ul><li> com atributos ARIA esperados', () => {
    const container = document.createElement('div');
    const clientes = [mockCliente({ id: 1 }), mockCliente({ id: 2, nome: 'Outro Cliente' })];
    const callbacks = { onEdit: jest.fn(), onDelete: jest.fn() };
    renderClientes(container, clientes, callbacks);
    const ul = container.querySelector('ul.cliente-list-ul');
    expect(ul).not.toBeNull();
    const itens = ul.querySelectorAll('li.cliente-card');
    expect(itens.length).toBe(2);
    expect(itens[0].getAttribute('role')).toBe('button');
    expect(itens[0].getAttribute('aria-expanded')).toBe('false');
    expect(itens[0].getAttribute('aria-pressed')).toBe('false');
    expect(container.innerHTML).toMatchSnapshot();
  });
});
