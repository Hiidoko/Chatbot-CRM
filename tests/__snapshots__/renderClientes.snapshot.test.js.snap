// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderClientes (estrutura semântica) gera <ul><li> com atributos ARIA esperados 1`] = `
"<ul class="cliente-list-ul" role="list"><li class="cliente-card collapsed" tabindex="0" role="button" aria-expanded="false" aria-pressed="false" data-status="novo">
      <div class="cliente-info">
        <strong class="field field-nome" data-field="nome">Cliente Teste</strong>
        <span class="field field-status" data-field="status"><i class="fa-solid fa-flag"></i> <strong class="status-badge status-novo">novo</strong></span>
        <span class="field pill" data-field="email"><i class="fa-solid fa-envelope"></i> cliente@teste.com</span>
        <span class="field pill" data-field="telefone"><i class="fa-solid fa-phone"></i> (11) 99999-9999</span>
        <span class="field pill" data-field="cidade"><i class="fa-solid fa-location-dot"></i> São Paulo</span>
        <span class="field pill" data-field="maquina"><i class="fa-solid fa-gears"></i> Máquina A</span>
        <span class="field pill" data-field="horario"><i class="fa-solid fa-clock"></i> Manhã</span>
        <span class="field pill" data-field="consultor"><i class="fa-solid fa-user-tie"></i> Consultor X</span>
        <span class="field meta" data-field="dataCadastro"><i class="fa-solid fa-calendar"></i> 31/12/2023</span>
      </div>
      <div class="cliente-actions">
        <button class="edit" data-id="1"><i class="fa-solid fa-pen-to-square"></i> Editar</button>
        <button class="delete" data-id="1"><i class="fa-solid fa-trash"></i> Excluir</button>
      </div><span class="cliente-toggle-indicator">+</span></li><li class="cliente-card collapsed" tabindex="0" role="button" aria-expanded="false" aria-pressed="false" data-status="novo">
      <div class="cliente-info">
        <strong class="field field-nome" data-field="nome">Outro Cliente</strong>
        <span class="field field-status" data-field="status"><i class="fa-solid fa-flag"></i> <strong class="status-badge status-novo">novo</strong></span>
        <span class="field pill" data-field="email"><i class="fa-solid fa-envelope"></i> cliente@teste.com</span>
        <span class="field pill" data-field="telefone"><i class="fa-solid fa-phone"></i> (11) 99999-9999</span>
        <span class="field pill" data-field="cidade"><i class="fa-solid fa-location-dot"></i> São Paulo</span>
        <span class="field pill" data-field="maquina"><i class="fa-solid fa-gears"></i> Máquina A</span>
        <span class="field pill" data-field="horario"><i class="fa-solid fa-clock"></i> Manhã</span>
        <span class="field pill" data-field="consultor"><i class="fa-solid fa-user-tie"></i> Consultor X</span>
        <span class="field meta" data-field="dataCadastro"><i class="fa-solid fa-calendar"></i> 31/12/2023</span>
      </div>
      <div class="cliente-actions">
        <button class="edit" data-id="2"><i class="fa-solid fa-pen-to-square"></i> Editar</button>
        <button class="delete" data-id="2"><i class="fa-solid fa-trash"></i> Excluir</button>
      </div><span class="cliente-toggle-indicator">+</span></li></ul>"
`;
